analyzer/src/analyze/analyzer.py

```diff
@@ -201,3 +193,27 @@ def get_primary_key(self, resource_mapping):
             resource_mapping["primaryKeyColumn"],
             resource_mapping["source"]["credential"]["owner"],
         )
+
+    @staticmethod
+    def get_analysis_columns(analysis):
+        analysis_columns = set()
+        for attribute in analysis.attributes:
+            for input_group in attribute.input_groups:
+                for col in input_group.columns:
+                    analysis_columns.add(col)
+
+                for condition in input_group.conditions:
+                    analysis_columns.add(condition.sql_column)
+
+        return analysis_columns
+
+    @staticmethod
+    def get_analysis_joins(analysis):
+        analysis_joins = set()
+        for attribute in analysis.attributes:
+            for input_group in attribute.input_groups:
+                for col in input_group.columns:
+                    for join in col.joins:
+                        analysis_joins.add(join)
```

Comment:
👍  Don't we also need to check the joins in the mapping are in the correct order ?

Reply:
This method is not used to build the sql query (you can see how it's done in QueryBuilder)
It is only used to build the squash rules and in some unit tests, I think
But I think it could use a refactor, I'm just not yet sure how to improve this part